diff --git a/dev/null b/arch/arm64/boot/dts/rockchip/rk3566-radxa-cm3-spa.dts
--- a/dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-cm3-spa.dts
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Rockchip Electronics Co., Ltd.
+ * Copyright (c) 2021 Radxa Limited.
+ *
+ */
+
+/dts-v1/;
+
+#include "rk3566-radxa-rock-3-compute-module.dtsi"
+
+/ {
+	model = "Radxa CM3 SPA";
+	compatible = "radxa,radxa-cm3-spa", "rockchip,rk3566";
+
+	vcc_sd: sdmmc-regulator {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio0 RK_PA5 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&sdmmc_pwren>;
+		regulator-name = "vcc_sd";
+		regulator-always-on;
+	};
+};
+
+&bus_npu {
+	status = "disabled";
+};
+
+&rknpu {
+	status = "disabled";
+};
+
+&rknpu_mmu {
+	status = "disabled";
+};
+
+&rknpu_reserved {
+	status = "disabled";
+};
+
+&i2s0_8ch {
+	status = "disabled";
+};
+
+&display_subsystem {
+    status = "disabled";
+};
+
+&hdmi {
+	status = "disabled";
+};
+
+&hdmi_in_vp0 {
+	status = "disabled";
+};
+
+&hdmi_in_vp1 {
+	status = "disabled";
+};
+
+&route_hdmi {
+	status = "disabled";
+};
+
+&sata1 {
+	status = "disabled";
+};
+
+&sata2 {
+	status = "disabled";
+};
+
+&gpio_leds {
+	pi-led-green {
+		gpios = <&gpio4 4 GPIO_ACTIVE_LOW>;
+		linux,default-trigger = "timer";
+		default-state = "on";
+		pinctrl-0 = <&pi_led>;
+	};
+
+	pwr-led-red {
+		gpios = <&gpio4 15 GPIO_ACTIVE_LOW>;
+		linux,default-trigger = "none";
+		default-state = "on";
+		pinctrl-0 = <&pwr_led>;
+	};
+};
+
+&sdmmc0 {
+	status = "disabled";
+};
+
+&pcie2x1 {
+	status = "disabled";
+};
+
+&pinctrl {
+	leds {
+		pi_led: pi-led {
+			rockchip,pins = <4 RK_PA4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		pwr_led: pwr-led {
+			rockchip,pins = <4 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	sdcard {
+		sdmmc_pwren: sdmmc-pwren {
+			rockchip,pins = <0 RK_PA5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
